pre_commit/commands/run.py

```diff
@@ -117,9 +117,14 @@ def _run_single_hook(classifier, hook, skips, cols, verbose, use_color):
 
         diff_cmd = ('git', 'diff', '--no-ext-diff')
         diff_before = cmd_output_b(*diff_cmd, retcode=None)
-        filenames = tuple(filenames) if hook.pass_filenames else ()
+        if hook.pass_diff:
+            inputs = (diff_before[1],)
```

Comment:
this isn't the correct value fwiw, `diff_before` is the *unstaged* diff (used to determine whether things change or not)